collectors/charts.d.plugin/nvme/nvme.chart.sh

```diff
@@ -0,0 +1,144 @@
+# shellcheck shell=bash
+# no need for shebang - this file is loaded from charts.d.plugin
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# netdata
+# real-time performance and health monitoring, done right!
+# (C) 2016 Costa Tsaousis <costa@tsaousis.gr>
+#
+# Made by Alexandre Azedo | SRE <aazedo@broadvoice.com>
+# 2022
+
+# _update_every is a special variable - it holds the number of seconds
+# between the calls of the _update() function
+nvme_update_every=5
+
+# the priority is used to sort the charts on the dashboard
+# 1 = the first chart
+nvme_priority=49000
+
+# global variables to store our collected data
+critical_warning=
+percentage_used=
+temperature=
+power_cycles=
+power_on_hours=
+
+nvme_get() {
+  # do all the work to collect / calculate the values
+  # for each dimension
+  #
+  # Remember:
+  # 1. KEEP IT SIMPLE AND SHORT
+  # 2. AVOID FORKS (avoid piping commands)
+  # 3. AVOID CALLING TOO MANY EXTERNAL PROGRAMS
+  # 4. USE LOCAL VARIABLES (global variables may overlap with other modules)
+  nvme_list=$(sudo -n nvme list | awk '/nvme/ { print $1 }')
+  local i=
+  for i in ${nvme_list[*]}; do
```

Comment:
Interesting, shellcheck doesn’t seem to like this either way...

I’m not enough of an expert on bash arrays to know whether we need the double quotes here or not, but either way we’re probably going to need a one-line override for the relevant check.

Assuming this works correctly as things are written right now, the suggested change attached to this comment can be used to silence the shellcheck warning.

```suggestion
  # shellcheck disable=SC2048
  for i in ${nvme_list[*]}; do
```

Reply:
This plugin has been working on my servers for months without problems. 
Thanks @Ferroin